src/dialog/enemyeditor.cpp

```diff
@@ -0,0 +1,1188 @@
+		TabRef(name = "Defenses", TabPanel(
+		))
+	);
+	auto flags_tab = TabPanel(
+		ptr = &guy_tabs[3],
+		TabRef(name = "Flags", TabPanel(
+		))
+	);
+	auto movement_tab = TabPanel(
+		ptr = &guy_tabs[4],
+		TabRef(name = "Movement", TabPanel(
+		))
+	);
+	auto attack_tab = TabPanel(
+		ptr = &guy_tabs[5],
+		TabRef(name = "Attack", TabPanel(
+		))
+	);
+	auto effects_tab = TabPanel(
+		ptr = &guy_tabs[6],
+		TabRef(name = "Effects", TabPanel(
+		))
+	);
+	auto scripts_tab = TabPanel(
+		ptr = &guy_tabs[7],
+		TabRef(name = "Scripts", TabPanel(
+		))
+	);
+	window = Window(
+		use_vsync = true,
+		title = titlebuf,
+		info = "Edit enemies, setting up their graphics, effects, and attributes.",
+		onClose = message::CANCEL,
+		shortcuts = {
+			//i actually dont know of any, perhaps add shortcut keys for navigation?
+		},
+		Column(
+			TabPanel(
+				TabRef(name = "Basics", basics_tab)
+				TabRef(name = "Defenses",defenses_tab),
+				TabRef(name = "Flags",flags_tab),
+				TabRef(name = "Movement",movement_tab),
+				TabRef(name = "Attack",attack_tab),
+				TabRef(name = "Effects",effects_tab),
+				TabRef(name = "Scripts",scripts_tab)
+			),
+			Row(
+				vAlign = 1.0,
+				spacing = 2_em,
+				Button(
+					focused = true,
+					text = "OK",
+					minwidth = 90_px,
+					onClick = message::OK),
+				warnbtn = Button(
+					text = "Warnings",
+					minwidth = 90_px,
+					onClick = message::WARNINGS),
+				Button(
+					text = "Clear",
+					minwidth = 90_px,
+					onClick = message::CLEAR),
+				Button(
+					text = "Default",
+					minwidth = 90_px,
+					onClick = message::DEFAULT),
+				Button(
+					text = "Cancel",
+					minwidth = 90_px,
+					onClick = message::CANCEL)
+			)
+		)
+	);
+	refreshScript();
+	updateWarnings();
+	return window;
+}
+
+void EnemyEditorDialog::apply_enemy()
+{
+	guysbuf[index] = local_guyref;
+	strncpy(guy_string[index], enemy_name.c_str(), 63);
+	saved = false;
+	edited = true;
+}
+
+bool EnemyEditorDialog::handleMessage(const GUI::DialogMessage<message>& msg)
+{
+	guydata blank;
+	switch (msg.message)
+	{
+	case message::CLEAR:
+	{
+		bool doclear = false;
+		AlertDialog("Are you sure?",
+			"Clearing the enemy will set all values to null!",
+			[&](bool ret, bool)
+			{
+				doclear = ret;
+			}).show();
+		if (doclear)
+		{
+			local_guyref = guysbuf[eNONE];
+			printf(guy_string[index], "e%d", index);
+			rerun_dlg = true;
+			return true;
+		}
+		return false;
+	}
+	case message::DEFAULT:
+	{
+		bool doclear = false;
+		AlertDialog("Are you sure?",
+			"This will reset all values back to their defaults!",
+			[&](bool ret, bool)
+			{
+				doclear = ret;
+			}).show();
+		if (doclear)
+		{
+			if (index < 177)
+			{
+				local_guyref = default_guys[index];
+				strncpy(guy_string[index], old_guy_string[index], strlen(old_guy_string[index]));
+			}
+			else
+			{
+				local_guyref = default_guys[eNONE]; //enemies above 176 dont have defaults!
+				sprintf(guy_string[index], "e%d", index);
```

Comment:
This will give `e0`,`e1`,...`e100`,`e101`. If you want it instead to give `e000`,`e001`,...`e100`,`e101`, you could use this variation:
```suggestion
				sprintf(guy_string[index], "e%03d", index);
```